### PR TITLE
feat(er): add new resource support flow log

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:
Add new resource support flow log.
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
 make testacc TEST=./huaweicloud/services/acceptance/er TESTARGS='-run TestAccFlowLog_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/er -v -run TestAccFlowLog_basic -timeout 360m -parallel 4
=== RUN   TestAccFlowLog_basic
=== PAUSE TestAccFlowLog_basic
=== CONT  TestAccFlowLog_basic
--- PASS: TestAccFlowLog_basic (210.44s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/er        210.495s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/er/resource_huaweicloud_er_flow_log_test.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+func testAccFlowLogImportStateFunc() resource.ImportStateIdFunc {
+	return func(s *terraform.State) (string, error) {
+		var instanceId, flowLogId string
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type == "huaweicloud_er_flow_log" {
+				instanceId = rs.Primary.Attributes["instance_id"]
+				flowLogId = rs.Primary.ID
+			}
+		}
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/er/resource_huaweicloud_er_flow_log_test.go]:
```suggestion
		rs, ok := s.RoorModule().Resources[rName]
		if !ok {
			return "", fmt.Errorf("resource (%s) not found: %s", rName, rs)
		}
		instanceId = rs.Primary.Attributes["instance_id"]
		flowLogId = rs.Primary.ID
```

File: huaweicloud/services/acceptance/er/resource_huaweicloud_er_flow_log_test.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+		if instanceId == "" || flowLogId == "" {
+			return "", fmt.Errorf("some import IDs are missing, want '<instance_id>/<flow_log_id>', but '%s/%s'",
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/er/resource_huaweicloud_er_flow_log_test.go]:
```suggestion
			return "", fmt.Errorf("some import IDs are missing, want '<instance_id>/<id>', but got '%s/%s'",
```

File: huaweicloud/services/acceptance/er/resource_huaweicloud_er_flow_log_test.go (hunk 1 above)
Comment (wuzhuanhong) [reply huaweicloud/services/acceptance/er/resource_huaweicloud_er_flow_log_test.go]: fixed

File: huaweicloud/services/acceptance/er/resource_huaweicloud_er_flow_log_test.go (hunk 2 above)
Comment (wuzhuanhong) [reply huaweicloud/services/acceptance/er/resource_huaweicloud_er_flow_log_test.go]: fixed

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/er TESTARGS='-run TestAccFlowLog_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/er -v -run TestAccFlowLog_basic -timeout 360m -parallel 4
=== RUN   TestAccFlowLog_basic
=== PAUSE TestAccFlowLog_basic
=== CONT  TestAccFlowLog_basic
--- PASS: TestAccFlowLog_basic (211.15s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/er        211.224s

File: huaweicloud/services/er/resource_huaweicloud_er_flow_log.go [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+		return fmt.Errorf("error %s flow log: %s", operateType, err)
+	}
+
+	return nil
+}
+
+func buildUpdateFlowLogBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":        utils.ValueIngoreEmpty(d.Get("name")),
+		"description": utils.ValueIngoreEmpty(d.Get("description")),
```
Comment (Lance52259) [inline huaweicloud/services/er/resource_huaweicloud_er_flow_log.go]: If the description can be updated with empty value, we shouldn't ingore the value.

Comment (wuzhuanhong) [reply huaweicloud/services/er/resource_huaweicloud_er_flow_log.go]: fixed

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/er TESTARGS='-run TestAccFlowLog_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/er -v -run TestAccFlowLog_basic -timeout 360m -parallel 4
=== RUN   TestAccFlowLog_basic
=== PAUSE TestAccFlowLog_basic
=== CONT  TestAccFlowLog_basic
--- PASS: TestAccFlowLog_basic (213.48s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/er        213.562s

File: huaweicloud/services/er/resource_huaweicloud_er_flow_log.go [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildCreateFlowLogBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"flow_log": map[string]interface{}{
+			"log_store_type": utils.ValueIngoreEmpty(d.Get("log_store_type")),
```
Comment (ShiChangkuo) [inline huaweicloud/services/er/resource_huaweicloud_er_flow_log.go]: `ValueIngoreEmpty` is unnecessary for required parameters.

File: huaweicloud/services/er/resource_huaweicloud_er_flow_log.go [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
```
Comment (ShiChangkuo) [inline huaweicloud/services/er/resource_huaweicloud_er_flow_log.go:22]: missing the used APIs

File: huaweicloud/services/er/resource_huaweicloud_er_flow_log.go [hunk 6]
```diff
@@ -0,0 +1,412 @@
...
+	return nil
+}
+
+func flowLogStatusRefreshFunc(d *schema.ResourceData, meta interface{}, targets []string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		resp, err := getFlowLogInfo(d, meta)
+		if err != nil {
+			if _, ok := err.(golangsdk.ErrDefault404); ok && len(targets) < 1 {
+				return resp, "COMPLETED", nil
```
Comment (ShiChangkuo) [inline huaweicloud/services/er/resource_huaweicloud_er_flow_log.go]: **DELETED** is more meaningful.
```suggestion
				return resp, "DELETED", nil
```

File: huaweicloud/services/er/resource_huaweicloud_er_flow_log.go [hunk 7]
```diff
@@ -0,0 +1,412 @@
...
+	}
+
+	return nil
+}
+
+func flowLogStatusRefreshFunc(d *schema.ResourceData, meta interface{}, targets []string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		resp, err := getFlowLogInfo(d, meta)
+		if err != nil {
+			if _, ok := err.(golangsdk.ErrDefault404); ok && len(targets) < 1 {
```
Comment (ShiChangkuo) [inline huaweicloud/services/er/resource_huaweicloud_er_flow_log.go]: if `targets` is only used here, I think we can define it as bool type

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/er TESTARGS='-run TestAccFlowLog_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/er -v -run TestAccFlowLog_basic -timeout 360m -parallel 4
=== RUN   TestAccFlowLog_basic
=== PAUSE TestAccFlowLog_basic
=== CONT  TestAccFlowLog_basic
--- PASS: TestAccFlowLog_basic (220.17s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/er        220.211s

File: huaweicloud/services/er/resource_huaweicloud_er_flow_log.go (hunk 7 above)
Comment (wuzhuanhong) [reply huaweicloud/services/er/resource_huaweicloud_er_flow_log.go]: fixed

File: huaweicloud/services/er/resource_huaweicloud_er_flow_log.go (hunk 6 above)
Comment (wuzhuanhong) [reply huaweicloud/services/er/resource_huaweicloud_er_flow_log.go]: fixed

File: huaweicloud/services/er/resource_huaweicloud_er_flow_log.go (hunk 5 above)
Comment (wuzhuanhong) [reply huaweicloud/services/er/resource_huaweicloud_er_flow_log.go:22]: fixed

File: huaweicloud/services/er/resource_huaweicloud_er_flow_log.go (hunk 4 above)
Comment (wuzhuanhong) [reply huaweicloud/services/er/resource_huaweicloud_er_flow_log.go]: fixed

**Pull request merged**